src/org/opendatakit/briefcase/ui/export/components/ConfigurationPanel.java

```diff
@@ -34,6 +34,7 @@
     configuration.ifEndDatePresent(form::setEndDate);
     configuration.ifPullBeforePresent(form::setPullBefore);
     configuration.ifPullBeforeOverridePresent(form::setPullBeforeOverride);
+    configuration.ifOverwriteExistingFilesPresent(form::setOverwriteExistingFiles);
```

Comment:
@ggalmazor I am getting some strange behavior when I include this line of code. The overwrite existing file check box comes checked by default. I'm not sure why but seems the initial configuration has overwrite existing files specified although I'm not sure how. I spent the whole of last night debugging and I tracked it down to this [emajj:MainBriefcaseWindow.java#L249](https://github.com/emajj/briefcase/blob/master/src/org/opendatakit/briefcase/ui/MainBriefcaseWindow.java#L249)
```java
exportPanel = ExportPanel.from(exportTerminationFuture, BriefcasePreferences.forClass(ExportPanel.class), appPreferences, BACKGROUND_EXECUTOR, analytics); 
```
The object returned by 
```java
BriefcasePreferences.forClass(ExportPanel.class)
```
Has overwriteExistingFiles. I placed a break point inside [emajj:BriefcasePreferences.java#L65-L68](https://github.com/emajj/briefcase/blob/master/src/org/opendatakit/briefcase/model/BriefcasePreferences.java#L65-L68)
```java
private BriefcasePreferences(Class<?> node, PreferenceScope scope) {
    this.node = node;
    this.preferences = scope.preferenceFactory(node);
  }
```
And when I evaluated preferences.get("overwriteExistingFiles", "false") after it has been initialized as seen in the constructor, it returns true instead of false.

As a result when ExportConfiguration.load(BriefcasePreferences) is called, overwriteExisting files is set to true by default.

[emajj:ExportConfiguration.java#L73-L82](https://github.com/emajj/briefcase/blob/master/src/org/opendatakit/briefcase/export/ExportConfiguration.java#L73-L82)
```java
public static ExportConfiguration load(BriefcasePreferences prefs) {
    return new ExportConfiguration(
        prefs.nullSafeGet(EXPORT_DIR).map(Paths::get),
        prefs.nullSafeGet(PEM_FILE).map(Paths::get),
        prefs.nullSafeGet(START_DATE).map(LocalDate::parse),
        prefs.nullSafeGet(END_DATE).map(LocalDate::parse),
        prefs.nullSafeGet(PULL_BEFORE).map(Boolean::valueOf),
        prefs.nullSafeGet(PULL_BEFORE_OVERRIDE).map(PullBeforeOverrideOption::from),
        prefs.nullSafeGet(OVERWRITE_EXISTING_FILES).map(Boolean::valueOf)
    );
  }
```
Not sure what is causing that. Using the debugger I have been able to place breakpoints in the code to get to the root cause but when I get here [emajj:BriefcasePreferences.java#L76](https://github.com/emajj/briefcase/blob/master/src/org/opendatakit/briefcase/model/BriefcasePreferences.java#L76)

I can't seem to figure out what is happening. Is there a configuration file somewhere that it's reading?

Reply:
I've been able to reproduce this problem... I'll see what I can find :P

Reply:
OK. I think the code is correct and it does what we need.

I've found that, like me, maybe you were changing the value of the checkbox in a configuration that was not valid. For the default conf to be valid, it has to have the export dir filled in. Invalid confs are not saved as user preferences and then, even though you are changing the value, it doesn't effectively change what was saved from the previous launch.

I've also reviewed the code on the listener and `requestOverwriteExistingFilesConfirmation()`. I think it could be more clear to have something like this:

```java
overwriteFilesField.addActionListener(__ -> {
  if (overwriteFilesField.isSelected() || (overwriteFilesField.isSelected() && confirmOverwriteFiles()))
    triggerOverwriteFiles();
});

// and then...

private boolean confirmOverwriteFiles() {
  if (showConfirmDialog(this, "Overwrite existing files?", "", YES_NO_OPTION) == YES_OPTION)
    return true;
  overwriteFilesField.setSelected(false);
  return false;
}
```

The changes here are:
- Simplified shorter names (not so important, but makes reading the code easier. Also, changed `xyzCheckbox to xyzField to mirror other field members)
- The decision to trigger is entirely made on the listener
- The `confirmOverwriteFiles()` method will always launch a confirm dialog. Before this, it was launching it conditionally, which can be unexpected.

Reply:
Thanks for looking into this. One question

Isn't  `overwriteFilesField.isSelected() || ((overwriteFilesField.isSelected() && confirmOverwriteFiles()) ` effectively the same as `overwriteFilesField.isSelected()`

I mean because of short circuiting, if overwriteFilesField.isSelected() is true, the other expressions will not be executed making the entire expression true. And if it's false the and in the second expression will make the second expression false making the entire expression false. Please correct me if I'm wrong.

Reply:
Sorry! I was missing a bang before the first member on the condition. I meant this:

```java
overwriteFilesField.addActionListener(__ -> {
  if (!overwriteFilesField.isSelected() || (overwriteFilesField.isSelected() && confirmOverwriteFiles()))
    triggerOverwriteFiles();
});
```

Reply:
Ah I c. Thanks.

Reply:
>The confirmOverwriteFiles() method will always launch a confirm dialog. Before this, it was launching it conditionally, which can be unexpected.

Looking at the code above. Not sure that will happen. If the action is unchecking the box, the the confirm dialog will not be shown. Which is what I was doing before.

What I was doing before is, if action is box checking, request confirmation. If confirmation is approval, call triggerOverwriteFiles(). But if action is unchecking, don't show dialog. Do you mean we should show the dialog even when the user is unchecking the box? But the code above will not achieve that because if user is unchecking the box,` overwriteFilesField.isSelected()` will return *false* and the *!* will make it *true* and due to to short circuit the other expression that contains the call to `confirmOverwriteFiles()` will not be called hence no confirmation box will be shown.

But the above your code looks shorter and cleaner. I just want to be sure I'm getting the expected behavior right.

Reply:
My idea was:
- If the user unchecks the checkbox, the confirm dialog gets shortcircuited (we don't show it), and we trigger the change
- If the user checks the checkbox, we show the confirm dialog
  - If the user confirms the action, trigger the change
  - If the user doesn't confirm the action, uncheck the checkbox and don't trigger the change

Now that I've reviewed it to write this text I realize that you're right. The condition could be:

```java
overwriteFilesField.addActionListener(__ -> {
  if (!overwriteFilesField.isSelected() || confirmOverwriteFiles())
    triggerOverwriteFiles();
});
```

A secondary goal for me was to make sure each time we execute `confirmOverwriteFiles()` a confirm dialog is shown (remove its conditional behavior).

Reply:
> secondary goal for me was to make sure each time we execute confirmOverwriteFiles() a confirm dialog is shown (remove its conditional behavior).

Now I get it. Yeah looking at my code again, it's a little misleading as it ask for confirmation based 
on a condition. Thanks for the help :)